client/src/components/admin_dashboard/ DataManagement.js

```diff
@@ -0,0 +1,137 @@
+import React,{useState, useEffect, Component} from 'react';
+import { Card, Search, Pagination, Grid, Segment, Button, Dropdown, Checkbox, Label, GridColumn } from 'semantic-ui-react'
+import { makeCall } from '../../apis';
+import MergeModal from './MergeModal'
+
+// import source data
+
+// define a fileter 
+const searchOptions = [
+    {
+        key:"All Fields",
+        text:"All Fields",
+        value:"all"
+    },
+    {
+        key:"Profession",
+        text:"Profession",
+        value:"jobTitleName"
+    },
+    {
+        key: 'Company',
+        text: 'Company',
+        value: 'companyName'
+    },
+    {
+        key: 'Major',
+        text: 'Major',
+        value: 'major'
+    },
+    {
+        key: 'Interest',
+        text: 'Interest',
+        value: 'interest'
+    }
+]
+
+const pageSize = 6;
+
+export default class DataManagement extends Component {
+    constructor(props){
+        super(props)
+        this.state={
+            allInterests : [],
+            checkedInterests : [],
+            sortedInterests : [],
+            display : [],
+            search : '',  
+            currPage : 1,
+            pageSize : 0,
+            mergeMoalOpen : false, 
+            mergeButtonActive : false,         
+        }
+        this.getInterests = this.getInterests.bind(this)
+        this.interestCard  = this.interestCard.bind(this)
+        this.handlePaginationChange =  this.handlePaginationChange.bind(this)
+    }
+
+    getInterests(){
+        makeCall({}, '/admin/allInterests/' + this.props.userDetails._id, 'get')
+        .then((res) =>{
+            this.setState(
+                {allInterests : res.interests})
+        })
+    };
+
+    componentWillMount(){
+        this.getInterests();
```

Comment:
this method is not done executing before this.state.allInterests is populated, since this.getInterests returns a promise. I would suggest converting getInterests to a async method (remove the .then in the method), and use an `await` here